### PR TITLE
Add a section about reactive in the JSON guide

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

**Review by FroMage — APPROVED** [review]
LGTM

File: docs/src/main/asciidoc/rest-json.adoc [hunk 1]
```diff
@@ -383,6 +383,47 @@ Let's do that and follow the same steps as before:
...
+    return findAll();
+}
+----
+
+Use `Uni` when you have a single result.
+Use `Multi` when you have multiple items that may be emitted asynchronously.
+
+You can use `Uni` and `Response` to return asynchronous HTTP responses: `Uni<Response>`.
+
+More details about Mutiny can be found on the link:getting-started-reactive#mutiny[Getting Started with Reactive guide].
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-json.adoc:424]:
```suggestion
More details about Mutiny can be found in the link:getting-started-reactive#mutiny[Getting Started with Reactive guide].
```

File: docs/src/main/asciidoc/rest-json.adoc [hunk 2]
```diff
@@ -383,6 +383,47 @@ Let's do that and follow the same steps as before:
 
 This time, you can see our list of legumes.
 
+[[reactive]]
+== Being reactive
+
+You can return _reactive types_ to handle asynchronous processing.
+Quarkus recommends the usage of https://smallrye.io/smallrye-mutiny[Mutiny] to write reactive and asynchronous code.
+
+To integrate Mutiny and RestEasy, you need to add the `quarkus-resteasy-mutiny` dependency to your project:
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest-json.adoc:392]:
```suggestion
To integrate Mutiny and RESTEasy, you need to add the `quarkus-resteasy-mutiny` dependency to your project:
```

**Review by gsmet — COMMENTED** [review]
Added 2 minor comments, feel free to ignore.

**Pull request merged**